packages/firestore/src/local/local_store.ts

```diff
@@ -175,6 +177,7 @@ export class LocalStore {
       this.mutationQueue,
       this.persistence.getIndexManager()
     );
+    this.queryEngine.setLocalDocumentsView(this.localDocuments);
```

Comment:
How come we construct the QueryEngine in FirestoreClient and pass it into LocalStore, rather than just construct it here in LocalStore?  If we could do that, then we wouldn't need this awkward `setLocalDocumentsView()` method that must be called before use...

Reply:
It allows to parameterize the tests for the LocalStore. Fun times.

Reply:
It's tempting to suggest that we just have a `useIndexFreeQueryEngine` boolean argument or something that tests can use...  But I don't care that much.